### DISCUSSION TITLE
ln: failed to create symbolic link. Protocol error

### DISCUSSION THREAD

**Discussion by claudiagp14** [gmtsar/gmtsar#940] [Q&A] [answered]
Hello everyone!

I am struggling with some problems with my workflow in GMTSAR. I am working with a virtual machine and I downloaded the Ubuntu Virtual machine that has GMTSAR (http://topex.ucsd.edu/gmtsar/tar/Ubuntu20.0_GMTSAR6.1.ova). I am following the PDF of: "How to make an InSAR time series from Sentinel-1 TOPS data: Kilauea", but with other images of other region.

The problem that I am dealing is this one, when I try to createt a symbolic link: 

gmtsar-user@gmtsarVB:/media/PhDSAR/gmtsar/F1/raw$ ln -s ../../data/*EOF .

gmtsar-user@gmtsarVB:/media/PhDSAR/gmtsar/F1/raw$ ln -s ../topo/dem.grd .

gmtsar-user@gmtsarVB:/media/PhDSAR/gmtsar/F1/raw$ ln -s ../../data/F*/*.SAFE/*/*iw1*hh*xml .

ln: failed to create symbolic link './*iw1*hh*xml': Protocol error

As you can see, I can make symbolic links with the DEM and the orbits, but I can't do it with the xmls. 

What I've done until now:
1. To get the permissions (cause before I had the problem of ln: failed to create symbolic link './*iw1*hh*xml': Operation not permitted): 
sudo visudo
gmtsar-user ALL=(ALL:ALL) ALL
2. I have also verified the mounting options with: 
mount | grep 
3. In the command prompt of Windows:
VBoxManage setextradata "Ubuntu21.04" VBoxInternal2/SharedFoldersEnableSymlinksCreate/PhDSAR 1
Clarification PhDSAR is my shared folder.

4. Run as an administrator the virtual machine

I have also to mention that I skipped the step 4c of the PDF, because it does not create a directory called Fxxxx_Fxxxx. Does anyone know which is the problem?

Thank you very much!

Claudia

**Comment by Xiaohua-Eric-Xu** [comment]
Hmmm, first time seeing this error. Is it because you don't have permissions at your present working directory? `4. Run as an administrator the virtual machine` should've fixed this problem but I am really not sure. 

If you could attach a log of your terminal output, that'll tell us a bit more on what happened.

**Reply by claudiagp14** [reply]
I think I know where the bug is. 

I read the PDF again, I performed step 4b and 4c. Step 4b worked for me (so I already have the orbits). But I tried the step 4c to organize the data in this directory (attached screenshot of the code):

organize_files_tops.csh SAFE_filelist ../reframed/pins.ll 1 &> oft_mode1.log &
organize_files_tops.csh SAFE_filelist ../reframed/pins.ll 2 &> oft_mode2.log &
tail -f oft_mode2.log

So when I run the following code: "ln -s ../.../data/F*/*.SAFE/*/*/*iw1*hh*xml ." it does not work because no F directory has been created. Do anyone could know what I should do?

I have also to mention that my polarization is hh

Many thanks


![image](https://github.com/gmtsar/gmtsar/assets/169824396/ebefb8cc-4fa8-4eb0-8a27-bc33cebafca0)
![Captura de pantalla 2024-07-02 113456](https://github.com/gmtsar/gmtsar/assets/169824396/4ecab84f-c56d-49e4-9df9-bbb64a6ea6cf)
[oft_mode1.log](https://github.com/user-attachments/files/16086533/oft_mode1.log)
[oft_mode2.log](https://github.com/user-attachments/files/16086534/oft_mode2.log)

**Reply by Xiaohua-Eric-Xu** [reply]
What's in your SAFE_filelist and pins.ll? It is strange that the the script got killed immediately at the start.

**Reply by claudiagp14** [reply]
Hi Eric, this is what I have (I had to put it in a ZIP, cause GITHUB does not let me to put it originally. 

[Send.zip](https://github.com/user-attachments/files/16107686/Send.zip)

**Reply by Xiaohua-Eric-Xu** [reply]
They look normal to me. How much memory (RAM) did you allocate for the virtual machine? Feels like it killed the program when running make_s1a_tops. You may try reproducing this error by running make_s1a_tops on one of your tiff and xml files inside your .SAFE file.

**Reply by claudiagp14** [reply]
Hi Eric,

The memory of the RAM is the one that is by default (8GB RAM and 50GB disk space allocated), that is on Github: http://topex.ucsd.edu/gmtsar/tar/Ubuntu20.0_GMTSAR6.1.ova

This is what it appears if I apply what you mentioned. 
![image](https://github.com/gmtsar/gmtsar/assets/169824396/d0cc4c19-6987-4984-ba0d-9e565f08042c)

**Reply by Xiaohua-Eric-Xu** [reply]
You need to feed arguments to make_s1a_tops to actually run the program. e.g., `make_s1a_tops ****.tiff ****.xml test 2`

**Reply by claudiagp14** [reply]
Hi Eric,
I just run it and this is what it appears:
![Captura de pantalla 2024-07-15 130027](https://github.com/user-attachments/assets/0911ec47-8104-48d0-8ea1-95ca3e16cec3)

**Reply by chintals** [reply, marked as answer]
The command string is incorrect. It should be `make_s1a_tops ****.xml ****.tiff test 2`  first xml and then tiff
